cmd/kubeadm/app/util/kubeletclient/client.go

```diff
@@ -0,0 +1,96 @@
+/*
+Copyright 2018 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package kubeletclient
+
+import (
+	"fmt"
+	"io/ioutil"
+	"net/http"
+
+	"k8s.io/api/apps/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	restclient "k8s.io/client-go/rest"
+	"k8s.io/kubernetes/pkg/api/legacyscheme"
+	"k8s.io/kubernetes/pkg/apis/core"
+	"k8s.io/kubernetes/pkg/kubelet/client"
+	kubetypes "k8s.io/kubernetes/pkg/kubelet/types"
+)
+
+// KubeletClient represents a Kublet Client
+type KubeletClient struct {
+	config    client.KubeletClientConfig
+	transport http.RoundTripper
+}
+
+// CreateClient creates a KubeletClient
+func CreateClient() (KubeletClient, error) {
+	kubeletClient := KubeletClient{}
+	kubeletClient.config = client.KubeletClientConfig{
+		Port:        10250,
+		EnableHttps: true,
+		TLSClientConfig: restclient.TLSClientConfig{
+			CertFile: "/etc/kubernetes/pki/apiserver-kubelet-client.crt",
+			KeyFile:  "/etc/kubernetes/pki/apiserver-kubelet-client.key",
+			Insecure: true,
```

Comment:
Would we re-use that cert or generate a kubeadm-client specific cert for the final implementation?

Reply:
Do we need to add the kubernetes ca crt to this so that it's not `Insecure: true`?

Reply:
I think it's fine to re-use this keypair, though I plan on updating it to pull the paths from the config rather than hardcoding.